db/migrate/20240228140211_create_data_migration.rb

```diff
@@ -0,0 +1,13 @@
+class CreateDataMigration < ActiveRecord::Migration[7.1]
+  def change
+    create_table :data_migrations do |t|
+      t.string :model, null: false
+      t.integer :processed_count, null: false, default: 0
```

Comment:
do we need a total of rows to process, although we can use a total of processed vs failure I guess, do we trust those numbers fully?

Reply:
I thought about this, but computing it adds complexity (especially for users, where we are only migrating a subset of the dataset) so I left it off for now. I think it may be worth adding if we find we want an more accurate % complete update in the UI, but I'm not sure if that's going to be useful yet so I left it off. We should be able to take `processed_count` as the total at the end of the migration (unless the migration itself fails but that would be evident elsewhere). The `failure_count` is a subset of `processed_count`.